### ISSUE TITLE
Signing in with Google Account not working

### ISSUE THREAD

**Issue by lucasstinis** [iMicknl/ha-nest-protect#277]
### The problem

When I press "authorize your account" during the integration's setup process, I receive an "Access blocked: Nest’s request is invalid" error from Google:

> You can’t sign in because Nest sent an invalid request. You can try again later, or contact the [developer](https://accounts.google.com/) about this issue. [Learn more about this error](https://support.google.com/accounts/answer/12379384)
If you are a developer of Nest, see [error details](https://accounts.google.com/).
Error 400: invalid_request_

### What version of this integration (ha-nest-protect) has the issue?

0.3.12

### What version of Home Assistant Core has the issue?

2023.11.3

### Home Assistant log

<details><summary>Logs</summary>
```
Copy/paste any log here, between the starting and ending backticks (`)
```

</details>

### Additional information

I've tried this both while already being logged in to a Google account in my browser, and while not being logged in, in both cases the error is thrown.

**Comment by royiversen78** [comment]
Just got a Nest Protect today, and have the same issue

**Comment by dmartyparty** [comment]
Snap... Got the same error here today.

**Comment by nunocavaco** [comment]
Same here, can't configure it!

**Comment by bigjunglecats** [comment]
I got the same error trying to sign in.

**Comment by jonbach** [comment]
Same issue

Access blocked: Nest’s request is invalid

You can’t sign in because Nest sent an invalid request. You can try again later, or contact the [developer](https://accounts.google.com/) about this issue. [Learn more about this error](https://support.google.com/accounts/answer/12379384)

If you are a developer of Nest, see [error details] (https://accounts.google.com/).

Error 400: invalid_request

**Comment by QuinnDamerell** [comment]
Same issue. Clicking "read details" gives this:

Error 400: invalid_request
The out-of-band (OOB) flow has been blocked in order to keep users secure. Follow the Out-of-Band (OOB) flow migration guide linked in the developer docs below to migrate your app to an alternative method.
Request details: redirect_uri=urn:ietf:wg:oauth:2.0:oob
[Related developer documentation](https://developers.google.com/identity/protocols/oauth2/resources/oob-migration)

It looks like the oauth system being used is no longer supported?

**Comment by iMicknl** [comment]
Duplicate of #105. Please see the pinned issue.

TLDR: use the beta branch. Hopefully I have more time soon to improve the beta branch... Currently documentation is limited.
